docs/sdks/js-reference-guide/js-sdk-redeemOtp.md

```diff
@@ -0,0 +1,69 @@
+---
+title: redeemOtp
+id: js-reference-redeemOtp
+description: ''
+slug: /js-reference-redeemOtp
+keywords:
+ - javascript sdk
+pagination_next: null
+pagination_prev: null
+last_update:
+   date: 07/25/2023
+   author: William May
+draft: false
+doc_type: reference
+displayed_sidebar: sdkSidebar
+---
+
+The **redeemOtp** function enables an app using the Beyond Identity Javascript SDK to redeem an otp for a grant code.
+
+## Dependencies
+
+The **redeemOtp** function requires the Beyond Identity Javascript SDK.
+```
+npm install @beyondidentity/bi-sdk-js
+```
+
+## Prerequisites
+
+Before making a call to **redeemOtp**, you must complete the following prerequisite calls:
+
+1. Import the required types and functions from the SDK
+```javascript
+import {Embedded} from '@beyondidentity/bi-sdk-js';
+```
+
+2. Initialize the SDK
+```javascript
+const embedded = await Embedded.initialize();
+```
+
+3. Use **redeemOtp** to redeem an otp for a grant code
+```javascript
+await embedded.redeemOtp(url, otp);
+```
+
+## Parameters
+
+| Parameter | Type | Description |
+| --- | --- | --- |
+| url | string | Required. The authentication URL of the current transaction. |
+| otp | string | Required. The OTP to redeem. |
+
+## Returns
+
+On success, the **redeemOtp** function returns a Promise that resolves to an **AuthenticateResponse**, which itself is a JSON object that contains the following keys:
+  - **redirectUrl**: string containing the redirect URL that originates from the /authorize call's `redirect_uri` parameter. The OAuth2 authorization `code` and the `state` parameter of the /authorize call are attached with the "code" and "state" parameters to this URL.
+  - **message**: optional string containing a displayable message defined by policy returned by the cloud on success.
+  - **passkeyBindingToken**: string containing a one-time-token that may be redeemed for a CredentialBindingLink.
```

Comment:
should we add detail  on how to do this? 

```
const response = await fetch(
    `${env(
      "FRONTGATE_API_BASE_URL"
    )}/tenants/${tenantId}/realms/${realmId}/applications/${applicationId}/credential-binding-jobs`,
    {
      method: "POST",
      headers: { Authorization: `Bearer ${passkeyBindingToken}` },
    }
  );
  ```

Reply:
Added.

Reply:
I think we want to put the actual `FRONTGATE_API_BASE_URL` here too.

Reply:
I can't find the URL in the production pipeline. Is this `https://api-us.beyondidentity.com/`? I think I'll leave /v1 since I'm pointing to the API docs anyway.

Reply:
I think it's `auth-us` but we should confirm. If we don't know then the developer definitely won't know how to find it!

Reply:
This was merged, but I made another [PR](https://github.com/gobeyondidentity/developer-docs/pull/299) to resolve this conversation.